--- a/src/anyp/Uri.cc
+++ b/src/anyp/Uri.cc
@@ -334,8 +334,11 @@
             SBufToCstring(foundHost, rawHost);
 
-            if (!tok.skip(':'))
-                throw TextException("missing required :port in CONNECT target", Here());
-            foundPort = parsePort(tok);
+            if (!tok.skip(':')) {
+                foundPort = 443;
+            } else {
+                (void)tok.skipAll(CharacterSet::WSP);
+                foundPort = tok.remaining().isEmpty() ? 443 : parsePort(tok);
+            }
 
             if (!tok.remaining().isEmpty())
                 throw TextException("garbage after host:port in CONNECT target", Here());
