--- squid-7.3.orig/src/ipcache.cc
+++ squid-7.3/src/ipcache.cc
@@ -323,10 +323,14 @@
 static ipcache_entry *
 ipcache_get(const char *name)
 {
-    if (ip_table != nullptr)
-        return (ipcache_entry *) hash_lookup(ip_table, name);
-    else
-        return nullptr;
+    if (ip_table != nullptr) {
+        char *newName = xstrdup(name);
+        Tolower(static_cast<char*>(newName));
+        ipcache_entry *entry = (ipcache_entry *) hash_lookup(ip_table, newName);
+        free(newName);
+        return entry;
+    }
+    return nullptr;
 }
 
 /// \ingroup IPCacheInternal
